src/lib/yang/binary.lua

```diff
@@ -244,6 +244,11 @@ local function data_emitter(production)
             stream:write_stringref('stringref')
             stream:write_stringref(data)
          end
+      elseif primitive_type == 'empty' then
+         return function (data, stream)
+            stream:write_stringref('stringref')
+            stream:write_stringref('')
```

Comment:
I see what you are saying -- that the "struct" emitter won't recurse into the leaf of type "empty" if the value is nil.  Apologies for misunderstanding and confusing the issue.  Still I agree with you that we should make sure that the value that a present leaf of type "empty" is always "true" and not just any true value.